src/utils/slack.test.js

```diff
@@ -0,0 +1,178 @@
+const {
+  addEmojiReaction,
+  getChannelID,
+  getSlackUsers,
+  getSlackUsersInConversation,
+  postEphemeralResponse,
+  postMessage,
+  sendDirectMessage,
+  setClient,
+} = require("./slack");
+
+describe("utils / slack", () => {
+  const defaultClient = {
+    chat: { postMessage: jest.fn() },
+    conversations: { list: jest.fn(), open: jest.fn() },
+    users: { list: jest.fn() },
+  };
+
+  beforeAll(() => {
+    setClient(defaultClient);
+    process.env.SLACK_TOKEN = "slack token";
+  });
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it("adds an emoji reaction to a message", async () => {
+    const msg = {
+      client: {
+        reactions: {
+          add: jest.fn(),
+        },
+      },
+      event: {
+        channel: "channel id",
+        ts: "message timestamp",
+      },
+    };
+
+    await addEmojiReaction(msg, ":bob:");
+
+    expect(msg.client.reactions.add).toHaveBeenCalledWith({
```

Comment:
I'm not sure I understand the question. Do you mean getting the object that was passed to `msg.client.reactions.add`? We _could_ get that with `msg.client.reactions.add.mock.calls[0][0]`.

Reply:
in general i prefer testing the state of objects rather than if a mocked function was called but i think i just misunderstood what was happening in this code trail! Thanks.